test/Microsoft.AspNet.Mvc.Core.Test/Routing/UrlHelperTest.cs

```diff
@@ -851,6 +857,210 @@ public void Action_RouteValueInvalidation_DoesNotAffectActionAndController_Actio
             Assert.Equal("/b/Store/Checkout", url);
         }
 
+        [Fact]
+        public void UrlAction_DefaultIgnoredForPrecidence()
+        {
+            // Arrange
+            var services = GetServices();
+            var router = BuildRouterWithAttributeRoute(services);
+            var urlHelper = CreateUrlHelper("/", router, services);
+
+            // Act
+            var url = urlHelper.Action(
+                action: "Default",
+                controller: "matchVariable");
+
+            // Assert
+            Assert.Equal("/api/MatchVariable/default", url);
+        }
+
+        [Fact]
+        public void UrlAction_DefaultUsedWhenSupplied()
+        {
+            // Arrange
+            var services = GetServices();
+            var router = BuildRouterWithAttributeRoute(services);
+            var urlHelper = CreateUrlHelper("/", router, services);
+
+            // Act
+            var url = urlHelper.Action(
+                action: "Default",
+                controller: "matchVariable",
+                values: new
+                {
+                    id = 1234
+                });
+
+            // Assert
+            Assert.Equal("/api/MatchVariable/default/1234", url);
+        }
+
+        [Fact]
+        public void UrlAction_MatchMostVariables()
+        {
+            // Arrange
+            var services = GetServices();
+
+            var router = BuildRouterWithAttributeRoute(services);
+
+            var urlHelper = CreateUrlHelper("/", router, services);
+
+            // Act
+            var url = urlHelper.Action(
+                action: "Get",
+                controller: "MatchVariable",
+                values: new { id = 1234 });
+
+            // Assert
+            Assert.Equal("/api/MatchVariable/help/1234", url);
+        }
+
+        [Fact]
+        public void UrlAction_WithoutVariables()
+        {
+            // Arrange
+            var services = GetServices();
+            var router = BuildRouterWithAttributeRoute(services);
+            var urlHelper = CreateUrlHelper("/", router, services);
+
+            // Act
+            var url = urlHelper.Action(
+                action: "Get",
+                controller: "MatchVariable");
+
+            // Assert
+            Assert.Equal("/api/MatchVariable/help", url);
+        }
+
```

Comment:
This is a good set of tests. Look at testing this more with the AttributeRoute directly - there are existing tests of this flavor, and it should be simpler.

I'd also like to see some tests with more variables.